### PR TITLE
Add isNotEmpty

### PR DESCRIPTION
**Pull request by electricRGB**
This adds an `isNotEmpty` property for `Collection` `String` and `Data` to improve  legibility of  code.

This is an easy review, but care must be taken when looking at each change. I've done two passes, so I think I changed everything correctly, but may have missed something, so please be careful

### CODE REVIEW

**Comment by lmarceau** [comment]
There's some errors on the BR:
Screen Shot 2022-02-24 at 1 17 59 PM

File: Client/Experiments/Settings/ExperimentsSettingsViewController.swift [hunk 1]
```diff
@@ -37,7 +37,7 @@ class ExperimentsSettingsViewController: UIViewController {
     }
 
     @objc private func updateState() {
-        experimentsView.reloadButton.isEnabled = !(experimentsView.customRemoteSettingsTextField.text?.isEmpty ?? true)
+        experimentsView.reloadButton.isEnabled = (experimentsView.customRemoteSettingsTextField.text?.isNotEmpty ?? true)
```
Comment (electricRGB) [inline Client/Experiments/Settings/ExperimentsSettingsViewController.swift:40]:
```suggestion
        experimentsView.reloadButton.isEnabled = !(experimentsView.customRemoteSettingsTextField.text?.isEmpty ?? true)
```

Comment (lmarceau) [reply Client/Experiments/Settings/ExperimentsSettingsViewController.swift:40]: Could be
`experimentsView.reloadButton.isEnabled = experimentsView.customRemoteSettingsTextField.text?.isNotEmpty ?? false` ?

Comment (gutley) [reply Client/Experiments/Settings/ExperimentsSettingsViewController.swift:40]: Personally I never like those `someOptionalBool ?? [true|false]` as they need some real thought to figure out what is happening in the nil case. I've come to preferring actually comparing the optional with the value you want:
```
experimentsView.reloadButton.isEnabled = (experimentsView.customRemoteSettingsTextField.text?.isNotEmpty == true)
```

I think early Swifts wouldn't allow that, but it's perfectly legal nowadays.

**Comment by electricRGB** [comment]
After some more research and debate, I've decided that this is not a useful PR and there are some great reasons for not actually doing this. Closing for now. :)

**Pull request closed without merging**